statistics/histogram.go

```diff
@@ -430,28 +430,17 @@ func (hg *Histogram) ToString(idxCols int) string {
 // equalRowCount estimates the row count where the column equals to value.
 // matched: return true if this returned row count is from Bucket.Repeat or bucket NDV, which is more accurate than if not.
 func (hg *Histogram) equalRowCount(value types.Datum, hasBucketNDV bool) (count float64, matched bool) {
-	index, match := hg.Bounds.LowerBound(0, &value)
-	// Since we store the lower and upper bound together, if the index is an odd number, then it points to a upper bound.
-	if index%2 == 1 {
-		if match {
-			return float64(hg.Buckets[index/2].Repeat), true
-		}
-		if hasBucketNDV && hg.Buckets[index/2].NDV > 1 {
-			return float64(hg.bucketCount(index/2)-hg.Buckets[index/2].Repeat) / float64(hg.Buckets[index/2].NDV-1), true
-		}
-		return hg.notNullCount() / float64(hg.NDV), false
+	exceed, bucketIdx, inBucket, match := hg.locateBucket(value)
+	if exceed || !inBucket {
```

Comment:
Should we need the first return value `exceed` necessary? Because when the `exceed = true`, the `inBucket` is always false. Can we just keep `inBucket` rather than both of `exceed` and `inBucket`.